From 53bc780bbf5be2da9c7f775d48b10b7e216ae688 Mon Sep 17 00:00:00 2001
From: Pooja Prajod <a0132412@ti.com>
Date: Tue, 5 May 2015 16:20:20 +0530
Subject: [PATCH] Solve segmentation fault in drm_handle_format

The format info as part of display is unnecessary. This patch removes
the code which for storing format info. Since format info was stored in an
array, there will not be  overflow pertaining to this.

Signed-off-by: Pooja Prajod <a0132412@ti.com>
---
 ext/wayland/gstwaylandsink.c | 8 --------
 ext/wayland/gstwaylandsink.h | 3 ---
 2 files changed, 11 deletions(-)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index 93a6bda..e450296 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -428,15 +428,7 @@ drm_handle_device (void *data, struct wl_drm *drm, const char *device)
 static void
 drm_handle_format (void *data, struct wl_drm *drm, uint32_t format)
 {
-  struct display *d = data;
   GST_DEBUG ("got format: %" GST_FOURCC_FORMAT, GST_FOURCC_ARGS (format));
-  if (d->drm_format_count < 50) {
-    d->drm_formats[d->drm_format_count++] = format;
-  } else {
-    d->drm_formats[d->drm_format_count++] = format;
-    GST_WARNING ("drm_handle_formats (no room for more formats): %"
-        GST_FOURCC_FORMAT, GST_FOURCC_ARGS (format));
-  }
 }
 
 static void
diff --git a/ext/wayland/gstwaylandsink.h b/ext/wayland/gstwaylandsink.h
index fee7c1c..d0ddf49 100644
--- a/ext/wayland/gstwaylandsink.h
+++ b/ext/wayland/gstwaylandsink.h
@@ -88,9 +88,6 @@ struct  display
   struct wl_drm *drm;
   uint32_t formats;
 
-  uint32_t drm_formats[50];
-  int drm_format_count;
-	
   struct wl_list input_list;
   int seat_version;
   uint32_t serial;
-- 
1.7.12.4

